### PR TITLE
feat(proxy): export errors

### PR DESCRIPTION
**Pull request by Halimao**
Partially addresses: #1140 

Exports errors for `proxy` package.

### CODE REVIEW

**Comment by faddat** [comment]
fyi your code is fine the test that failed is known-bad

File: proxy/errors.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+func (e ErrNewProxyClient) Error() string {
+	return fmt.Sprintf("failed to connect to proxy: %v", e.Err)
+}
+
+func (e ErrNewProxyClient) Unwrap() error {
+	return e.Err
+}
+
+type ErrABCIClient struct {
```
Comment (melekes) [inline proxy/errors.go:19]:
```suggestion
type ErrABCIClientCreate struct {
```

File: proxy/multi_app_conn.go [hunk 2]
```diff
@@ -151,7 +151,7 @@ func (app *multiAppConn) startConsensusClient() error {
 func (app *multiAppConn) startClient(c abcicli.Client, conn string) error {
 	c.SetLogger(app.Logger.With("module", "abci-client", "connection", conn))
 	if err := c.Start(); err != nil {
-		return fmt.Errorf("error starting ABCI client (%s client): %w", conn, err)
+		return ErrABCIClient{Action: "starting", CliType: conn, Err: err}
```
Comment (melekes) [inline proxy/multi_app_conn.go:154]: I would extract this into a separate `ErrABCIClientStart`

**Review by melekes — COMMENTED** [review]
Thanks @Halimao ❤️ minor comments about naming

**Comment by Halimao** [comment]
> Thanks @Halimao ❤️ minor comments about naming

@melekes Suggestion changes addressed~

**Review by melekes — APPROVED** [review]
👍

File: proxy/errors.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+func (e ErrUnreachableProxy) Error() string {
+	return fmt.Sprintf("failed to connect to proxy: %v", e.Err)
+}
+
+func (e ErrUnreachableProxy) Unwrap() error {
+	return e.Err
+}
+
+type ErrABCIClientCreate struct {
+	CliType string
```
Comment (melekes) [inline proxy/errors.go:20]:
```suggestion
	ClientName string
```

Comment (Halimao) [reply proxy/errors.go]: @melekes Addressed~

**Review by melekes — APPROVED** [review]
👍

**Review by cason — APPROVED** [review]

**Pull request merged**